### PR TITLE
Get unit tests working with Okta SAML Response

### PR DESCRIPTION
**Pull request by kworam-rally**
When Zuul calls SAMLUtils.loadResource() or SAMLUtils. readClasspathResource() and either of them fails with an exception, there is no indication of this in Zuul.  After spending hours trying to track this down with print statements in Zuul, I added the code in this PR to print the exceptions to stdout before  throwing them to the caller.

### CODE REVIEW

File: src/main/java/com/rallydev/saml/SAMLUtils.java [hunk 1]
```diff
@@ -189,32 +189,49 @@ public static Credential toCredential(X509Certificate certificate) {
...
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
             }
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
+        } catch (Throwable t) {
+            // When Zuul calls this method and it fails with an exception, there is no indication of this in Zuul.
+            // After spending hours trying to track this down with print statements in Zuul,
+            // I added this code to print exceptions to stdout before throwing them to the caller.
+            t.printStackTrace(System.out);
```
Comment (ryeash) [inline src/main/java/com/rallydev/saml/SAMLUtils.java]: this isn't the right way to do this, if the exception is being suppressed in zuul, fix the problem in zuul, never use `printStackTrace`

File: src/main/java/com/rallydev/saml/SAMLUtils.java [hunk 2]
```diff
@@ -189,32 +189,49 @@ public static Credential toCredential(X509Certificate certificate) {
...
+                } else {
+                    return Files.readAllBytes(new File(uri.getPath()).toPath());
+                }
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
             }
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
+        } catch (Throwable t) {
+            // When Zuul calls this method and it fails with an exception, there is no indication of this in Zuul.
```
Comment (nedgladstone-rally) [inline src/main/java/com/rallydev/saml/SAMLUtils.java]: Having these comments about silence in zuul is probably a good idea, but maybe reduce them to a single line each.
// print exception stack trace to stdout to enable troubleshooting in zuul
or some such

File: src/main/java/com/rallydev/saml/SAMLUtils.java (hunk 1 above)
Comment (kworam-rally) [reply src/main/java/com/rallydev/saml/SAMLUtils.java]: fixed

File: src/main/java/com/rallydev/saml/SAMLUtils.java (hunk 2 above)
Comment (kworam-rally) [reply src/main/java/com/rallydev/saml/SAMLUtils.java]: n/a due to above fix

**Comment by ryeash** [comment]
:ship:

**Pull request merged**